analyses/molecular-subtyping-chordoma/01-Subtype-chordoma.Rmd

```diff
@@ -0,0 +1,231 @@
+  mutate(status = "neutral") %>%
+  # let's get the columns to match chordoma_loss
+  rename(biospecimen_id = Kids_First_Biospecimen_ID) %>%
+  select(biospecimen_id, status)
+copy_neutral_df
+```
+
+```{r}
+# remove large copy number data frame
+rm(focal_cn_df)
+```
+
+```{r}
+chordoma_copy <- chordoma_loss %>% 
+  #join the losses with the neutrals to get a new data frame
+  select(biospecimen_id, status) %>%
+  bind_rows(copy_neutral_df)
+chordoma_copy
+```
+
+Need to get the sample_id that corresponds to biospecimen_id into chordoma_copy so we can match WGS and RNA-seq biospecimens from the same event/sample:
+```{r}
+chordoma_copy <- chordoma_copy %>%
+  # get only the Kids_First_Biospecimen_ID, sample_id columns from our identifier data.frame
+  # then use biospecimen IDs to add the sample_id info
+  inner_join(select(chordoma_id_df,
+                    Kids_First_Biospecimen_ID,
+                    sample_id),
+             by = c("biospecimen_id" = "Kids_First_Biospecimen_ID"))
+chordoma_copy
+```
+
+Look at SMARCB1 expression values only in chordoma
+
+```{r}
+# get the row that contains the SMARCB1 values
+# gene symbols are rownames
+smarcb1_expression <- expression_data[which(rownames(expression_data) == "SMARCB1"), ]
+
+# now only the columns correspond to chordoma samples
+smarcb1_expression <- smarcb1_expression[, which(colnames(expression_data) %in% chordoma_samples) ]
+smarcb1_expression
+```
+
+```{r}
+# remove large expression matrix that's no longer needed
+rm(expression_data)
+```
+
+The `smarcb1_expression` is a not a friendly form ^^; Transposing needed: 
+
+```{r}
+# transpose such that samples are rows
+smarcb1_expression <- t(smarcb1_expression) %>%
+  # make a data.frame
+  as.data.frame() %>%
+  # we want the rownames that are biospecimen identifers as their own column called Kids_First_Biospecimen_ID
+  tibble::rownames_to_column("Kids_First_Biospecimen_ID") %>%
+  # give SMARCB1 column a slightly better column name
+  rename(SMARCB1_expression = SMARCB1)
+smarcb1_expression
+```
+This also needs sample_id to add it in:
+
+```{r}
+smarcb1_expression <- smarcb1_expression %>%
+  inner_join(select(chordoma_id_df,
+                    Kids_First_Biospecimen_ID,
+                    sample_id),
+             by = "Kids_First_Biospecimen_ID")
+smarcb1_expression
+```
+
+```{r}
+chordoma_smarcb1_df <- smarcb1_expression %>%
+  # any missing samples will get filled with NA when using a full join
+  full_join(chordoma_copy, by = "sample_id")
+chordoma_smarcb1_df
+```
+
+```{r}
+# this step adds in the participant identifier (sample_id to match between the two data.frame)
+chordoma_smarcb1_df <- chordoma_smarcb1_df %>%
+  inner_join(distinct(select(chordoma_id_df, 
+                             sample_id, 
+                             Kids_First_Participant_ID)),
+             by = "sample_id")
+
+# combining the two biospecimen identifiers to a single column (all biospecimen IDs for a sampl separated by a comma)
```

Comment:
Possibly -  I suppose this was a step to make sure the Kids_First_Participant_ID is clearly the way to identify/select by?
@jaclyn-taroni - could you comment on this

Reply:
We started out handling the biospecimen IDs this was for the reason that @mkoptyra points out. But since giving it more thought and realizing the clinical file this will go into is focused on the Kids_First_Biospecimen_ID, we now do as @jashapiro suggestions in the HGG (#249) and non-MB, non-ATRT embryonal tumors. I can change this part @mkoptyra and then leave a detailed comment about what I did. Let me know when you’ve pushed your changes and are ready for me to make edits!

Reply:
Now we rename the columns `Kids_First_Biospecimen_ID_DNA` and `Kids_First_Biospecimen_ID_RNA` -- 
```r
chordoma_smarcb1_df <- smarcb1_expression %>%
  # any missing samples will get filled with NA when using a full join
  full_join(chordoma_copy, by = "sample_id") %>%
  rename(Kids_First_Biospecimen_ID_DNA = Kids_First_Biospecimen_ID,
         Kids_First_Biospecimen_ID_RNA = biospecimen_id)
```
That's what the `rename` function is doing.

I also now do:
```r
chordoma_smarcb1_df <- chordoma_id_df %>%
  select(sample_id, Kids_First_Participant_ID) %>%
  distinct() %>%
  inner_join(chordoma_smarcb1_df,
             by = "sample_id")
```

instead of:
```
chordoma_smarcb1_df <- chordoma_smarcb1_df %>%
  inner_join(distinct(select(chordoma_id_df, 
                             sample_id, 
                             Kids_First_Participant_ID)),
             by = "sample_id")
```

It's doing the same thing (adding the `Kids_First_Participant_ID` using the `chordoma_id_df` data frame) but it's a little easier to read in my opinion.